ext/stackprof/stackprof.c

```diff
@@ -593,9 +597,17 @@ stackprof_record_sample_for_stack(int num, uint64_t sample_timestamp, int64_t ti
     }
 }
 
+// buffer the current profile frames
+// This must be async-signal-safe
+// Returns immediately if another set of frames are already in the buffer
 void
-stackprof_record_sample()
+stackprof_buffer_sample(void)
 {
+    if (_stackprof.buffer_count > 0) {
+	// Another sample is already pending
+	return;
```

Comment:
Should we record the number of times this happened?  It's technically a missed sample I think.

Reply:
yep! We already have that information via `overall_signals` vs `overall_samples`